### PR TITLE
feat: add BitVec.ofBool_[and|or|xor]_ofBool theorems

### PR DESCRIPTION
**Pull request by tobiasgrosser**
... and use them to simplify some proofs.

### CODE REVIEW

**Review by hargoniX — APPROVED** [review]

File: src/Init/Data/BitVec/Lemmas.lean [hunk 1]
```diff
@@ -234,6 +234,15 @@ theorem ofBool_eq_iff_eq : ∀ {b b' : Bool}, BitVec.ofBool b = BitVec.ofBool b'
...
 @[simp] theorem not_ofBool : ~~~ (ofBool b) = ofBool (!b) := by cases b <;> rfl
 
+@[simp] theorem ofBool_and_ofBool : ofBool b &&& ofBool b' = ofBool (b && b') := by
+  cases b <;> cases b' <;> rfl
+
+@[simp] theorem ofBool_or_ofBool : ofBool b ||| ofBool b' = ofBool (b || b') := by
+  cases b <;> cases b' <;> rfl
+
+@[simp] theorem ofBool_of_ofBool : ofBool b ^^^ ofBool b' = ofBool (b ^^ b') := by
+  cases b <;> cases b' <;> rfl
```
Comment (tobiasgrosser) [inline src/Init/Data/BitVec/Lemmas.lean:244]:
```suggestion
@[simp] theorem ofBool_xor_ofBool : ofBool b ^^^ ofBool b' = ofBool (b ^^ b') := by
  cases b <;> cases b' <;> rfl
```

File: tests/lean/interactive/completionPrv.lean.expected.out [hunk 2]
```diff
@@ -46,6 +46,30 @@
...
+    "id": {"const": {"declName": "BitVec.ofBool_and_ofBool"}}}},
+  {"sortText": "5",
+   "label": "BitVec.ofBool_of_ofBool",
+   "kind": 3,
+   "data":
+   {"params":
+    {"textDocument": {"uri": "file:///completionPrv.lean"},
+     "position": {"line": 9, "character": 11}},
+    "id": {"const": {"declName": "BitVec.ofBool_of_ofBool"}}}},
+  {"sortText": "6",
```
Comment (tobiasgrosser) [inline tests/lean/interactive/completionPrv.lean.expected.out:64]:
```suggestion
    "id": {"const": {"declName": "BitVec.ofBool_xor_ofBool"}}}},
  {"sortText": "6",
```

File: tests/lean/interactive/completionPrv.lean.expected.out [hunk 3]
```diff
@@ -46,6 +46,30 @@
...
+   "label": "BitVec.ofBool_and_ofBool",
+   "kind": 3,
+   "data":
+   {"params":
+    {"textDocument": {"uri": "file:///completionPrv.lean"},
+     "position": {"line": 9, "character": 11}},
+    "id": {"const": {"declName": "BitVec.ofBool_and_ofBool"}}}},
+  {"sortText": "5",
+   "label": "BitVec.ofBool_of_ofBool",
+   "kind": 3,
```
Comment (tobiasgrosser) [inline tests/lean/interactive/completionPrv.lean.expected.out:58]:
```suggestion
   "label": "BitVec.ofBool_or_ofBool",
   "kind": 3,
```

File: tests/lean/interactive/completionPrv.lean.expected.out [hunk 4]
```diff
@@ -46,6 +46,30 @@
...
+    "id": {"const": {"declName": "BitVec.ofBool_and_ofBool"}}}},
+  {"sortText": "5",
+   "label": "BitVec.ofBool_or_ofBool",
+   "kind": 3,
+   "data":
+   {"params":
+    {"textDocument": {"uri": "file:///completionPrv.lean"},
+     "position": {"line": 9, "character": 11}},
+    "id": {"const": {"declName": "BitVec.ofBool_xor_ofBool"}}}},
+  {"sortText": "6",
```
Comment (tobiasgrosser) [inline tests/lean/interactive/completionPrv.lean.expected.out:64]:
```suggestion
    "id": {"const": {"declName": "BitVec.ofBool_or_ofBool"}}}},
  {"sortText": "6",
```

File: tests/lean/interactive/completionPrv.lean.expected.out [hunk 5]
```diff
@@ -46,6 +46,30 @@
...
+   "label": "BitVec.ofBool_or_ofBool",
+   "kind": 3,
+   "data":
+   {"params":
+    {"textDocument": {"uri": "file:///completionPrv.lean"},
+     "position": {"line": 9, "character": 11}},
+    "id": {"const": {"declName": "BitVec.ofBool_or_ofBool"}}}},
+  {"sortText": "6",
+   "label": "BitVec.ofBool_or_ofBool",
+   "kind": 3,
```
Comment (tobiasgrosser) [inline tests/lean/interactive/completionPrv.lean.expected.out:66]:
```suggestion
   "label": "BitVec.ofBool_xor_ofBool",
   "kind": 3,
```

File: tests/lean/interactive/completionPrv.lean.expected.out [hunk 6]
```diff
@@ -46,6 +46,30 @@
...
+    "id": {"const": {"declName": "BitVec.ofBool_or_ofBool"}}}},
+  {"sortText": "6",
+   "label": "BitVec.ofBool_xor_ofBool",
+   "kind": 3,
+   "data":
+   {"params":
+    {"textDocument": {"uri": "file:///completionPrv.lean"},
+     "position": {"line": 9, "character": 11}},
+    "id": {"const": {"declName": "BitVec.ofBool_or_ofBool"}}}},
+  {"sortText": "7",
```
Comment (tobiasgrosser) [inline tests/lean/interactive/completionPrv.lean.expected.out:72]:
```suggestion
    "id": {"const": {"declName": "BitVec.ofBool_xor_ofBool"}}}},
  {"sortText": "7",
```

**Comment by tobiasgrosser** [comment]
@hargoniX, the tests pass.

**Pull request merged**